src/ethereum/fees/ethMiningFees.ts

```diff
@@ -277,3 +277,32 @@ function flatMap<T>(items: NestedArray<T>, destinationItems: T[] = []): T[] {
   })
   return destinationItems
 }
+
+export type GasParams =
+  | { gasPrice: string }
+  | {
+      maxPriorityFeePerGas: string
+      maxFeePerGas: string
+    }
+
+export async function getFeeParamsByTransactionType(
+  type: number,
```

Comment:
This variable name, `type`, doesn't give much info. Can we rename it, or put a comment saying "The type comes from as `TransactionFactory.fromTxData`" or such.